### ISSUE TITLE
Can't close Loqui

### ISSUE THREAD

**Issue by ghost** [loqui/im#426]
I can't close Loqui. The message "Closing the app will be closed in a few seconds" appears but Loqui does not close.

```
"Disconnect sequence initiated... ~ methodList.disconnect@app://{d06b8390-3257-4ff8-9feb-5e8dd4bbb316}/scripts/anonymous/coseme.js:8360 [ConnectionManager]" coseme.js:3539
"Sending term signal to reader thread ~ methodList.disconnect@app://{d06b8390-3257-4ff8-9feb-5e8dd4bbb316}/scripts/anonymous/coseme.js:8361 [ConnectionManager]" coseme.js:3539
"Disconnected! undefined ~ methodList.disconnect@app://{d06b8390-3257-4ff8-9feb-5e8dd4bbb316}/scripts/anonymous/coseme.js:8368 [ConnectionManager]" coseme.js:3539
"Error invoking disconnect with ["undefined"] ~ call@app://{d06b8390-3257-4ff8-9feb-5e8dd4bbb316}/scripts/anonymous/coseme.js:6656 [yowsupAPI]" coseme.js:3539
"Error details: TypeError: "_signalInterface is undefined" at .sendDisconnected@app://{d06b8390-3257-4ff8-9feb-5e8dd4bbb316}/scripts/anonymous/coseme.js:7558 > methodList.disconnect@app://{d06b8390-3257-4ff8-9feb-5e8dd4bbb316}/scripts/anonymous/coseme.js:8370 > call@app://{d06b8390-3257-4ff8-9feb-5e8dd4bbb316}/scripts/anonymous/coseme.js:6654 > @app://{d06b8390-3257-4ff8-9feb-5e8dd4bbb316}/scripts/loqui/connectors/coseme.js:53 > App.killAll@app://{d06b8390-3257-4ff8-9feb-5e8dd4bbb316}/scripts/loqui/app.js:237 > Menu.map.powerOff/req.onsuccess@app://{d06b8390-3257-4ff8-9feb-5e8dd4bbb316}/scripts/loqui/menu.js:272 >  ~ call@app://{d06b8390-3257-4ff8-9feb-5e8dd4bbb316}/scripts/anonymous/coseme.js:6657 [yowsupAPI]" coseme.js:3539
TypeError: _signalInterface is undefined coseme.js:7558
```

Loqui IM v0.3.1a / Firefox OS 1.2.0.0 / hamachi

**Comment by ghost** [comment]
After removing the XMPP account with the problems of #427 and #426 Loqui is shutting down normally

**Comment by ghost** [comment]
I can't shutdown Loqui IM, and it seems to be problem with the accounts again, after removing my Whatsapp account, it is possible to shutdown Loqui IM again

```
TypeError: _signalInterface is undefined coseme.js:7558
"Disconnect sequence initiated... ~ methodList.disconnect@app://a5c24eb4-b626-4fae-9f28-7d0a7f998310/scripts/anonymous/coseme.js:8360 [ConnectionManager]" coseme.js:3539
"Sending term signal to reader thread ~ methodList.disconnect@app://a5c24eb4-b626-4fae-9f28-7d0a7f998310/scripts/anonymous/coseme.js:8361 [ConnectionManager]" coseme.js:3539
"Disconnected! undefined ~ methodList.disconnect@app://a5c24eb4-b626-4fae-9f28-7d0a7f998310/scripts/anonymous/coseme.js:8368 [ConnectionManager]" coseme.js:3539
"Error invoking disconnect with ["undefined"] ~ call@app://a5c24eb4-b626-4fae-9f28-7d0a7f998310/scripts/anonymous/coseme.js:6656 [yowsupAPI]" coseme.js:3539
"Error details: TypeError: "_signalInterface is undefined" at .sendDisconnected@app://a5c24eb4-b626-4fae-9f28-7d0a7f998310/scripts/anonymous/coseme.js:7558 > methodList.disconnect@app://a5c24eb4-b626-4fae-9f28-7d0a7f998310/scripts/anonymous/coseme.js:8370 > call@app://a5c24eb4-b626-4fae-9f28-7d0a7f998310/scripts/anonymous/coseme.js:6654 > @app://a5c24eb4-b626-4fae-9f28-7d0a7f998310/scripts/loqui/connectors/coseme.js:53 > App.killAll@app://a5c24eb4-b626-4fae-9f28-7d0a7f998310/scripts/loqui/app.js:237 > Menu.map.powerOff/req.onsuccess@app://a5c24eb4-b626-4fae-9f28-7d0a7f998310/scripts/loqui/menu.js:272 >  ~ call@app://a5c24eb4-b626-4fae-9f28-7d0a7f998310/scripts/anonymous/coseme.js:6657 [yowsupAPI]"
```

Loqui IM v0.3.1 (Github) / Firefox OS 1.2.0.0 / hamachi

**Comment by aesedepece** [comment]
Sorry, accidentally closed the issue. We are still looking for a fix for this.

**Comment by MarkTr** [comment]
Is that related to #73 and #75?

**Comment by ghost** [comment]
@MarkTr I think this issue is not related to these old issues, did it happened to you?

It didn't happened to me since some time, maybe it is fixed

**Comment by aesedepece** [comment]
This still happens every time an unhandled error occurs, because it stops the JavaScript interpreter.

**Comment by ghost** [comment]
And suddenly:

```
TypeError: _signalInterface is undefined coseme.js:7559
"Disconnect sequence initiated... ~ methodList.disconnect@app://{b9ea79c1-addf-4760-ba8b-befd3e6f1dea}/scripts/anonymous/coseme.js:8361 [ConnectionManager]" coseme.js:3539
"Sending term signal to reader thread ~ methodList.disconnect@app://{b9ea79c1-addf-4760-ba8b-befd3e6f1dea}/scripts/anonymous/coseme.js:8362 [ConnectionManager]" coseme.js:3539
"Disconnected! undefined ~ methodList.disconnect@app://{b9ea79c1-addf-4760-ba8b-befd3e6f1dea}/scripts/anonymous/coseme.js:8369 [ConnectionManager]" coseme.js:3539
"Error invoking disconnect with ["undefined"] ~ call@app://{b9ea79c1-addf-4760-ba8b-befd3e6f1dea}/scripts/anonymous/coseme.js:6656 [yowsupAPI]" coseme.js:3539
"Error details: TypeError: "_signalInterface is undefined" at .sendDisconnected@app://{b9ea79c1-addf-4760-ba8b-befd3e6f1dea}/scripts/anonymous/coseme.js:7559 > methodList.disconnect@app://{b9ea79c1-addf-4760-ba8b-befd3e6f1dea}/scripts/anonymous/coseme.js:8371 > call@app://{b9ea79c1-addf-4760-ba8b-befd3e6f1dea}/scripts/anonymous/coseme.js:6654 > @app://{b9ea79c1-addf-4760-ba8b-befd3e6f1dea}/scripts/loqui/connectors/coseme.js:53 > App.killAll@app://{b9ea79c1-addf-4760-ba8b-befd3e6f1dea}/scripts/loqui/app.js:242 > Menu.map.powerOff/req.onsuccess@app://{b9ea79c1-addf-4760-ba8b-befd3e6f1dea}/scripts/loqui/menu.js:272 >  ~ call@app://{b9ea79c1-addf-4760-ba8b-befd3e6f1dea}/scripts/anonymous/coseme.js:6657 [yowsupAPI]"
```

**Comment by ghost** [comment]
The last error only happens when a Whatsapp account was added. If i remove the account again, i could close Loqui IM without problems.

**Comment by cmeerw** [comment]
fixed by commit 6ba959b307ee193ac33345a878b58fcbc03cf98a
